File: src/test/java/dev/bothin/testcontainers/java/Test2.java
package dev.bothin.testcontainers.java;

import dev.bothin.testcontainers.TestContainerConfigurationClass;
import dev.bothin.testcontainers.TestContainerLifecycleExtension;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import static org.junit.jupiter.api.Assertions.assertEquals;

@ExtendWith(TestContainerLifecycleExtension.class)
@TestContainerConfigurationClass(ContainerConfiguration.class)
class Test2 {
    @BeforeEach
    void setUp() {
        System.out.println("Java Test2 before");
    }

    @Test
    void when1() {
        System.out.println("Java Test2 when 1");
        assertEquals(1, 1);
    }


    @Test
    void when2() {
        System.out.println("Java Test2 when 2");
        assertEquals(2, 2);
    }
}

File: src/test/kotlin/dev/bothin/testcontainers/Test2.kt
package dev.bothin.testcontainers

import io.kotlintest.shouldBe
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(TestContainerLifecycleExtension::class)
@TestContainerConfigurationClass(ContainerConfiguration::class)
class Test2 {

    @BeforeEach
    fun setUp() {
        println("Test2 before")
    }

    @Test
    fun `when 1`() {
        println("Test2 when 1")
        1 shouldBe 1
    }


    @Test
    fun `when 2`() {
        println("Test2 when 2")
        2 shouldBe 2
    }
}